### PR TITLE
[CIS-1582] Fix `ChatMentionSuggestionView` permanently hiding subviews

### PR DESCRIPTION
**Pull request by evsaev**
### 🔗 Issue Link

CIS-1582

### 🎯 Goal

Fix issue in `ChatMentionSuggestionView` when once hidden subview was never shown again. This happens when users without the `name` are suggested.

### 🛠 Implementation

Manipulate `isHidden` depending on the text that is displayed.

### 🎨 Changes

https://user-images.githubusercontent.com/12818985/154441646-6b103806-cfaf-457d-b342-d0ecbb4a9870.mp4

### 🧪 Testing

Manual testing:
- Run `DemoApp`
- Create a group channel where one user does not have `name`
- Type `@` in the channel multiple times

See that the `name` label is hidden only for the user without `name` and always shown for other.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

**Review by nuno-vieira — DISMISSED** [review]
LGTM! ✅

File: Sources/StreamChatUI/CommonViews/Suggestions/ChatMentionSuggestionView/ChatMentionSuggestionView_Tests.swift [hunk 1]
```diff
@@ -50,15 +50,15 @@ class ChatMentionSuggestionView_Tests: XCTestCase {
         let view = ChatMentionSuggestionView().withoutAutoresizingMaskConstraints
         view.components = .mock
         view.widthAnchor.constraint(equalToConstant: Self.defaultCellWidth).isActive = true
-
+        
+        view.content = chatUserNoName
+        AssertSnapshot(view, suffix: "user name not set")
+
```
Comment (nuno-vieira) [inline Sources/StreamChatUI/CommonViews/Suggestions/ChatMentionSuggestionView/ChatMentionSuggestionView_Tests.swift:56]: Can we create a new testing case to make sure this issue doesn't comeback?

Comment (evsaev) [reply Sources/StreamChatUI/CommonViews/Suggestions/ChatMentionSuggestionView/ChatMentionSuggestionView_Tests.swift]: ✅ Added the test

**Review by nuno-vieira — APPROVED** [review]
LGTM! ✅

**Pull request merged**